src/librustc/infer/error_reporting/need_type_info.rs

```diff
@@ -100,7 +100,7 @@ impl<'a, 'gcx, 'tcx> InferCtxt<'a, 'gcx, 'tcx> {
         let mut labels = vec![(
             span,
             if &name == "_" {
-                "cannot infer type".to_string()
+                "cannot infer type".to_owned()
```

Comment:
There's no preference nowadays between `to_string` and `to_owned` other than style as both now have the same performance impact.

Reply:
You mean when https://github.com/rust-lang/rust/pull/54613 is merged? At this moment `to_string` is not inlined.